### ISSUE TITLE
Current version of Guard crashing with nil:Nilclass

### ISSUE THREAD

**Issue by pitosalas** [guard/guard#937]
The current version of guard, 2.15.1 running with Ruby 2.6.2 on the new MacOS Mojave is crashing as follows:

```
 guard
/Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/guard-2.15.1/lib/guard/commander.rb:56:in `stop': undefined method `stop' for nil:NilClass (NoMethodError)
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/guard-2.15.1/lib/guard/commander.rb:52:in `ensure in start'
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/guard-2.15.1/lib/guard/commander.rb:52:in `start'
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/guard-2.15.1/lib/guard/cli/environments/valid.rb:16:in `start_guard'
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/guard-2.15.1/lib/guard/cli.rb:122:in `start'
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/guard-2.15.1/lib/guard/aruba_adapter.rb:32:in `execute'
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/guard-2.15.1/lib/guard/aruba_adapter.rb:19:in `execute!'
	from /Users/pitosalas/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/guard-2.15.1/bin/_guard-core:11:in `<main>'
~/m/c/c/content (master *☡=)
```

**Comment by rykann** [comment]
I got the same error with Guard 2.15.1, Ruby 2.5.3, Mojave. As far as I can tell, the file change listener isn't getting initialized before something attempts to call it.

**Comment by pitosalas** [comment]
I downgraded to the previous gem version of Guard and the problem went away…

Pito Salas
Brandeis Computer Science
Volen 134

**Comment by rymai** [comment]
@pitosalas Hi, I can reproduce this if I run Guard without a Guardfile. Could you please share the content of your Guardfile?

**Comment by pitosalas** [comment]
Pito Salas
Brandeis Computer Science
Volen 134

**Comment by rymai** [comment]
Ok, I can reproduce it with a non-empty Guardfile. The problem is that when an error happens during the Guardfile evaluation, `stop` is always called but the actual error can happen before the listener is even created and the error gets swallowed by the `undefined method `stop' for nil:NilClass` error...

@pitosalas @rykann Could you please share the content of your Guardfile and tell me if you're running `guard` or `bundle exec guard`?

In the meantime, I'll push a quick-fix to not swallow the actual error.

**Comment by rymai** [comment]
I released Guard 2.16.1 which shouldn't swallow any `Guardfile` evaluation error that could happen.
